src/three-components/SmoothControls.ts

```diff
@@ -339,9 +369,28 @@ export class SmoothControls extends EventDispatcher {
     const {theta, phi, radius} = this[$targetSpherical];
 
     const targetTheta = theta - deltaTheta;
-    const targetPhi = phi - deltaPhi;
+    let targetPhi = phi - deltaPhi;
     const targetRadius = radius + deltaRadius;
 
+    const {
+      // minimumAzimuthalAngle: minTheta,
+      // maximumAzimuthalAngle: maxTheta,
+      minimumPolarAngle: minPhi,
+      maximumPolarAngle: maxPhi,
+    } = this[$options];
+
+    let phiScale = 1;
+
+    if (targetPhi < minPhi!) {
+      phiScale = 1 - Math.min((targetPhi - minPhi!) / (this[$minOverSpherical].phi - minPhi!), 1);
+    }
+
+    if (targetPhi > maxPhi!) {
+      phiScale = 1 - Math.min((targetPhi - maxPhi!) / (this[$maxOverSpherical].phi - maxPhi!), 1);
+    }
+
+    targetPhi = phi - deltaPhi * phiScale;
```

Comment:
Basically all this is doing is slowing down `deltaPhi` based on how far we potentially would have gone over the max and min values.